exporter/metric/metric.go

```diff
@@ -0,0 +1,384 @@
+	ctx := o.Context
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	client, err := monitoring.NewMetricClient(ctx, clientOpts...)
+	if err != nil {
+		return nil, err
+	}
+	e := &metricExporter{
+		client: client,
+		o:      o,
+	}
+	return e, nil
+}
+
+// InstallNewPipeline instantiates a NewExportPipeline and registers it globally.
+func InstallNewPipeline(popts ...push.Option) (*push.Controller, error) {
+	pusher, err := NewExportPipeline(popts...)
+	if err != nil {
+		return nil, err
+	}
+	global.SetMeterProvider(pusher)
+	return pusher, err
+}
+
+// NewExportPipeline sets up a complete export pipeline with the recommended setup,
+// chaining a NewRawExporter into the recommended selectors and integrators.
+func NewExportPipeline(popts ...push.Option) (*push.Controller, error) {
+	selector := simple.NewWithExactMeasure()
+	exporter, err := NewRawExporter()
+	if err != nil {
+		return nil, err
+	}
+	integrator := integrator.New(selector, true)
+	period := exporter.metricExporter.o.ReportingInterval
+	pusher := push.New(integrator, exporter, period, popts...)
+	return pusher, nil
+}
+
+// ExportMetrics exports OpenTelemetry Metrics to Google Cloud Monitoring.
+func (me *metricExporter) ExportMetrics(ctx context.Context, res *resource.Resource, cps export.CheckpointSet) error {
+	if err := me.exportMetricDescriptor(ctx, cps); err != nil {
+		return err
+	}
+
+	if err := me.exportTimeSeries(ctx, res, cps); err != nil {
+		return err
+	}
+	return nil
+}
+
+// exportMetricDescriptor create MetricDescriptor from the record
+// if the descriptor is not registerd in Cloud Monitoring yet.
+func (me *metricExporter) exportMetricDescriptor(ctx context.Context, cps export.CheckpointSet) error {
+	mds := make(map[string]*googlemetricpb.MetricDescriptor)
+	aggError := cps.ForEach(func(r export.Record) error {
+		desc := r.Descriptor()
+		name := desc.Name()
+
+		if _, ok := me.mdCache[name]; !ok {
+			if _, localok := mds[name]; !localok {
+				md := me.recordToMdpb(&r)
+				mds[name] = md
+			}
+		}
+		return nil
+	})
+	if aggError != nil {
+		return aggError
+	}
+	if len(mds) == 0 {
+		return nil
+	}
+
+	for _, md := range mds {
+		req := &monitoringpb.CreateMetricDescriptorRequest{
+			Name:             fmt.Sprintf("projects/%s", me.o.ProjectID),
+			MetricDescriptor: md,
+		}
+		_, err := me.client.CreateMetricDescriptor(ctx, req)
+		if err != nil {
+			return err
+		}
+		me.mdCache[md.Name] = md
+	}
+	return nil
+}
+
+// exportTimeSeriees create TimeSeries from the records in cps.
+// res should be the common resource among all TimeSeries, such as instance id, application name and so on.
+func (me *metricExporter) exportTimeSeries(ctx context.Context, res *resource.Resource, cps export.CheckpointSet) error {
+	tsCache := make(map[string][]*monitoringpb.TimeSeries)
+
+	aggError := cps.ForEach(func(r export.Record) error {
+		desc := r.Descriptor()
+		name := desc.Name()
```

Comment:
You only use `desc` here, so can just inline it as
```
name := r.Descriptor().Name()
```

Reply:
I did same in exportMetricDescriptor, so fixed both.